### PR TITLE
IR-198: Add default storage type for Nutanix platform

### PR DESCRIPTION
**Pull request by thunderboltsid**
Set default storage type for Nutanix platform to removed.

### CODE REVIEW

File: pkg/storage/storage.go [hunk 1]
```diff
@@ -183,6 +183,11 @@ func GetPlatformStorage(listers *regopclient.Listers) (imageregistryv1.ImageRegi
 	case configapiv1.AlibabaCloudPlatformType:
 		cfg.OSS = &imageregistryv1.ImageRegistryConfigStorageAlibabaOSS{}
 		replicas = 2
+	case configapiv1.NutanixPlatformType:
+		cfg.PVC = &imageregistryv1.ImageRegistryConfigStoragePVC{
```
Comment (dmage) [inline pkg/storage/storage.go:187]: We don't recommend to use PVC on new platforms. It's better to have the image registry removed by default (like we have for bare metal) or to use object storage.

Comment (thunderboltsid) [reply pkg/storage/storage.go:187]: Thanks for the review @dmage. Can you explain the rationale for the recommendation?

**Comment by dmage** [comment]
/assign @bmcelvee @xiuwang @sferich888 
for approvals

**Comment by xiuwang** [comment]
Can't install nutanix cluster so far, I will keep close with installer qe.

**Comment by xiuwang** [comment]
Verified the image registry operator default to Removed with pr build image.
/label qe-approved

**Comment by bmcelvee** [comment]
Note: no new content needs to be developed, but the _registry-removed_ module should be included in the installation guide. 

/label docs-approved

**Pull request merged**